sklearn/preprocessing/data.py

```diff
@@ -250,13 +313,65 @@ def fit(self, X, y=None):
         if feature_range[0] >= feature_range[1]:
             raise ValueError("Minimum of desired feature range must be smaller"
                              " than maximum. Got %s." % str(feature_range))
-        data_min = np.min(X, axis=0)
-        data_range = np.max(X, axis=0) - data_min
-        data_range = _handle_zeros_in_scale(data_range)
-        self.scale_ = (feature_range[1] - feature_range[0]) / data_range
-        self.min_ = feature_range[0] - data_min * self.scale_
-        self.data_range = data_range
-        self.data_min = data_min
+
+        if sparse.issparse(X):
+            raise TypeError("MinMaxScaler does no support sparse input.")
```

Comment:
Maybe suggest to use `MaxAbsScaler` that can scale sparse data without breaking the sparsity structure.